client/client.go

```diff
@@ -0,0 +1,208 @@
+package client
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/heptio/sonobuoy/pkg/config"
+	"github.com/heptio/sonobuoy/pkg/discovery"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+)
+
+// TODO(chuckha): These consts won't work for 0.9 and 0.8 tarballs. Make these less constant and more function.
+const (
+	// These consts define where files live in a v0.10 tarball. This will change to support other versions.
+
+	HostsDir                  = "hosts"
+	NamespacedResourcesDir    = "resources/ns"
+	NonNamespacedResourcesDir = "resources/cluster"
+	PluginsDir                = "plugins"
+	PodLogs                   = "podlogs"
+	MetadataDir               = "meta"
+	ServerVersionFile         = "serverversion.json"
+	ServerGroupsFile          = "servergroups.json"
+)
+
+// TODO(chuckha): export results
+
+// results holds a pointer to a file that contains Sonobuoy Results.
+type results struct {
```

Comment:
suggestion: make this an io.ReadCloser not an *os.file. It will mean that you won't be able to reopen or rewind the file pointer to create a second pass, but it would make it easier to test, and easier to support results that do not exists on a filesystem.